### PR TITLE
Spar: Forward errors from brig

### PR DESCRIPTION
**Pull request by smatting**
Fixes https://github.com/zinfra/backend-issues/issues/1613

### CODE REVIEW

**Comment by smatting** [comment]
All integration tests pass locally

File: services/spar/src/Spar/Error.hs [hunk 1]
```diff
@@ -188,3 +179,31 @@ renderSparError (SAML.CustomError SparProvisioningTokenLimitReached) = Right $ W
...
+  throwError err
+  where
+    err :: SparError
+    err =
+      responseJsonMaybe resp
+        & maybe
+          ( SAML.CustomError
+              . SparCouldNotParseRfcResponse serviceName
+              . ("internal error: " <>)
+              . cs
```
Comment (fisx) [inline services/spar/src/Spar/Error.hs:206]:
```suggestion
              . cs
              . show
              . (statusCode resp,)
```

(Or did you drop that on purpose?)

File: services/spar/src/Spar/Intra/Galley.hs [hunk 2]
```diff
@@ -27,45 +27,39 @@ import Data.Aeson (FromJSON, eitherDecode')
...
-parseResponse :: forall a m. (FromJSON a, MonadError SparError m, Typeable a) => ResponseLBS -> m a
+parseResponse :: (FromJSON a, MonadError SparError m) => ResponseLBS -> m a
 parseResponse resp = do
-  bdy <- maybe (throwSpar SparNoBodyInBrigResponse) pure $ responseBody resp
-  either err pure $ eitherDecode' bdy
-  where
-    err = throwSpar . SparCouldNotParseBrigResponse . (typeinfo <>) . cs
-    typeinfo = cs $ show (typeRep ([] @a)) <> ": "
+  bdy <- maybe (throwSpar $ SparCouldNotParseRfcResponse "galley" "no body") pure $ responseBody resp
+  either (throwSpar . SparCouldNotParseRfcResponse "galley" . cs) pure $ eitherDecode' bdy
```
Comment (fisx) [inline services/spar/src/Spar/Intra/Galley.hs]: I'd prefer to keep the `typeinfo`.  I'll make a commit to that end.

**Review by fisx — APPROVED** [review]
Nice cleanup!  I made a few small commits, and I think logging everything in rethrow with level `Error` is a mistake.  The other two comments are safe to ignore.  :-)

Please take another look and merge if you're happy.

**Comment by fisx** [comment]
integration tests passed locally.

File: services/spar/src/Spar/Error.hs (hunk 1 above)
Comment (smatting) [reply services/spar/src/Spar/Error.hs:206]: No, I probably deleted that on accident. Re-added

**Comment by smatting** [comment]
Locally `testUserUpdate` from `brig`'s integration test keep failing. Need to investigate..

**Pull request merged**